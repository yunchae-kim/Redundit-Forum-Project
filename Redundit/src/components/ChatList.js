import { useState, useEffect } from "react";
import { getAllChatgroups } from "../utility/axiosChatMessage";
import "../style/components/ChatList.css";
import "../style/pages/Global.css";

const ChatList = ({ userId, setActiveChat }) => {
  const [chatGroup, setChatGroup] = useState([]);

  const [isFolded] = useState(true);

  useEffect(async () => {
    const res = await getAllChatgroups(userId);
    setChatGroup(res.data);
  }, [userId]);

  const clickHandler = (event) => {
    const chatID = event.target.value;
    setActiveChat(chatID);
  };

  return (
    <div className={isFolded ? "global-right-list" : "global-right-list"}>
      <div className="CL-header">
        <div className="CL-title">ChatList</div>
      </div>
      <hr className="CL-hr" />
      {chatGroup && chatGroup.length > 0
        && chatGroup.map((group) => (
          <div key={group.idchat}>
            <div className="CL-entry">
              <button type="submit" className="CL-entry-name" value={group.idchat} onClick={clickHandler}>
                {group.name}
              </button>
            </div>
            <hr className="CL-hr" />
          </div>
        ))}
    </div>
  );
};

export default ChatList;
