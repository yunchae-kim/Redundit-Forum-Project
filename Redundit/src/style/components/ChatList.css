.ChatList {
  }
  
.ChatList-expand {
}

.CL-header {
  padding-top: 24px;
  padding-bottom: 24px;
  margin: 0px;
  font-size: 28px;
  font-weight: 600;
  color: #188bd8;
}

.CL-title {
  margin-top: 0px;
  margin-bottom: 0px;
}

.CL-entry {
  display: flex;
  justify-items: flex-start;
  align-items: center;
  font-size: 20px;
  margin-top: 16px;
  margin-bottom: 16px;
  text-align: left;
}

.CL-hr {
  border: 1px solid #d1e8f7;
  margin-bottom: 0px;
  margin-top: 0px;
}

.CL-userIcon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  align-self: center;
  margin-left: 32px;
}

.CL-entry-admin {
  color: #465771;
  margin-left: 32px;
}

.CL-entry-name {
  margin-left: 24px;
  color: #111;
}

.CL-btn-entry {
  display: block;
  font-size: 20px;
}

.CL-btn {
  color: #5da271;
  padding: 15px;
  font-size: 20px;
  font-weight: 600;
  background-color: transparent;
  border: none;
}
  