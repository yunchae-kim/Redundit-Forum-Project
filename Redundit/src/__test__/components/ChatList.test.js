import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import ChatList from "../../components/ChatList";

const mockAxios = new MockAdapter(axios);
mockAxios.onGet().reply(200, {
  data: [
    {
      idchat: 1,
      iduser: 1,
      name: "One",
      timestamp: {},
      private: true,
    },
    {
      idchat: 2,
      iduser: 1,
      name: "Two",
      timestamp: {},
      private: true,
    },
    {
      idchat: 3,
      iduser: 1,
      name: "Three",
      timestamp: {},
      private: true,
    }],
});

test("test should render ChatList component", async () => {
  render(<ChatList userId={1} />);
  await waitFor(() => {
    const label = screen.getByText(/ChatList/i);
    expect(label).toBeInTheDocument();
  });
});

test("test should ChatList match snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <ChatList userId={1} />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
