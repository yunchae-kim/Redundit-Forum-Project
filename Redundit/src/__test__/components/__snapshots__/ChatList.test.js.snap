// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should ChatList match snapshot 1`] = `
<div
  className="global-right-list"
>
  <div
    className="CL-header"
  >
    <div
      className="CL-title"
    >
      ChatList
    </div>
  </div>
  <hr
    className="CL-hr"
  />
</div>
`;
